sql/type.go

```diff
@@ -731,7 +743,18 @@ func (t datetimeT) Convert(v interface{}) (interface{}, error) {
 	case string:
 		t, err := time.Parse(DatetimeLayout, value)
```

Comment:
This is a little convoluted. Better to loop over the accepted layouts a single time, returning as soon as you find one that parses.